Add VolumeStorer API to the storage port layer

fdawg4l opened the pull request:
Includes a VolumeStorer interface definition to store/retrieve/remove
volumes.

Implements a cache implementation of the interface so as to avoid
unnecessary datastore operations.

This code isn't being called yet but is in support of #915 #916 and #917.

caglar10ur commented:
LGTM :sheep: :it:

dougm commented on lib/portlayer/storage/volume_cache.go:
```diff
@@ -0,0 +1,144 @@
+	var i int
+	l := make([]*Volume, len(v.vlc))
+	for _, vol := range v.vlc {
+		// this is idiotic
```
Could just return a '[]Volume' instead?  In any case, can you use the range index (currently ignored '_'), rather than the 'var i' and 'i++' ?  Or even drop the index and use 'l = append(l, &e)'

fdawg4l replied:
`v.vlc` is a map so the `_` is the key.  We can't use that as an index.  But i *can* use `append`.  I'll go ahead and do that.  Thanks!

dougm commented:
lgtm

seems like we could make VolumesList a little happier though

fdawg4l commented:
@hickeng i think i addressed all of the comments you had in my branch if you want to take a look.  The only issue is that now we're stuck behind a lock for every call but i guess that's unavoidable.

fdawg4l commented:
:recycle:

fdawg4l commented:
:hankey:

caglar10ur commented:
LGTM :sheep: :it:

fdawg4l commented:
.
